src/ir/LocalGraph.cpp

```diff
@@ -152,21 +181,21 @@ struct Flower : public CFGWalker<Flower, Visitor<Flower>, Info> {
           // we have gone through this block; now we must handle flowing to
           // the inputs
           if (curr->in.empty()) {
-            if (curr == entry) {
+            if (curr == currentBlock) {
```

Comment:
it looks like currentBlock only refers to the `entry`? In that case, perhaps the name could be changed to reflect that (`entryFlowBlock` perhaps)? Otherwise, this change seems surprising, it makes sense before with `entry`, but after the check is on the "current block" which seems like something not related to the entry.